### ISSUE TITLE
False Postive: default numeric fallback emits warning for any literals on either side of an operator

### ISSUE THREAD

**Issue by Apersoma** [rust-lang/rust-clippy#16744]
### Summary

Even when the type of the other side is known, there is a warning emitted. 

### Reproducer

```rust
#![warn(clippy::default_numeric_fallback)]
let a = 92i64;
// could be any operation between `a` and a literal, except `literal << a` and
// `literal >> a` because that should emit the warning.
let _ = a + 3;

let b = 92f64;
let _ = b + 3.0;
```

### Version

```text

```

**Comment by Apersoma** [comment]
I accidently hit ctrl+enter before i was done
